planning/autoware_static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp

```diff
@@ -188,4 +217,140 @@ std::vector<TrajectoryPoint> OptimizationTrajectoryBasedCenterline::optimize_tra
+  route_ptr->header.frame_id = "map";
+  route_ptr->set__start_pose(start_pose);
+  route_ptr->set__goal_pose(goal_pose);
+  route_ptr->set__segments(route_sections);
+  return route_ptr;
+}
+
+std::shared_ptr<autoware::path_generator::PathGenerator>
+OptimizationTrajectoryBasedCenterline::create_path_generator_node(
+  const geometry_msgs::msg::Pose current_pose, LaneletMapBin::ConstSharedPtr & map_bin_ptr,
+  std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute> route_ptr) const
+{
+  const auto path_generator_node =
+    std::make_shared<autoware::path_generator::PathGenerator>(create_node_options());
+
+  // set data in path_generator
+  autoware::path_generator::PathGenerator::InputData path_generator_input;
+  path_generator_input.lanelet_map_bin_ptr = map_bin_ptr;
+  path_generator_input.odometry_ptr = utils::convert_to_odometry(current_pose);
+  path_generator_input.route_ptr = route_ptr;
+  path_generator_node->set_planner_data(path_generator_input);
+
+  return path_generator_node;
+}
+
+std::shared_ptr<autoware::behavior_path_planner::PlannerData>
+OptimizationTrajectoryBasedCenterline::create_behavior_path_planner_data(
+  std::shared_ptr<RouteHandler> & route_handler_ptr) const
+{
+  // create planner_data
+  std::shared_ptr<autoware::behavior_path_planner::PlannerData> planner_data_ =
+    std::make_shared<autoware::behavior_path_planner::PlannerData>();
+  planner_data_->route_handler = route_handler_ptr;
+  planner_data_->parameters.refine_goal_search_radius_range = refine_goal_search_radius_range;
+  return planner_data_;
+}
+
+std::shared_ptr<autoware::behavior_path_planner::DefaultFixedGoalPlanner>
+OptimizationTrajectoryBasedCenterline::create_fixed_goal_planner(
+  const PathWithLaneId & raw_path_with_lane_id) const
+{
+  // create behavior_path_planner
+  std::shared_ptr<autoware::behavior_path_planner::DefaultFixedGoalPlanner> behavior_path_planner =
+    std::make_shared<autoware::behavior_path_planner::DefaultFixedGoalPlanner>();
+
+  autoware::behavior_path_planner::BehaviorModuleOutput behavior_module_input_data;
+  behavior_module_input_data.path = raw_path_with_lane_id;
+  behavior_path_planner->setPreviousModuleOutput(behavior_module_input_data);
+
+  return behavior_path_planner;
+}
+
+path_generator::Params OptimizationTrajectoryBasedCenterline::create_params(
+  std::shared_ptr<autoware::path_generator::PathGenerator> & path_generator_node) const
+{
+  // create params
+  auto param_listener_ = std::make_shared<::path_generator::ParamListener>(
+    path_generator_node->get_node_parameters_interface());
+  return param_listener_->get_params();
+}
+
+PathWithLaneId OptimizationTrajectoryBasedCenterline::modify_goal_connection(
+  const PathWithLaneId & raw_path_with_lane_id, const Path & raw_path,
+  std::shared_ptr<RouteHandler> & route_handler_ptr, LaneletMapBin::ConstSharedPtr & map_bin_ptr,
+  const Pose & current_pose, const Pose & goal_pose) const
+{
+  if (goal_method == "None") {
```

Comment:
Define goal_method variable here with `getOrDeclareParameter` and remove the member variable.